pkg/util/hardware/hw_utils.go

```diff
@@ -59,6 +101,65 @@ func ParseCPUSetLine(cpusetLine string) (cpusList []int, err error) {
 	return
 }
 
+func ParseCPUSetFile(file string) ([]int, error) {
+	line, err := util.ScanLine(file)
+	if err != nil {
+		return nil, err
+	}
+	cpusList, err := ParseCPUSetLine(line)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse cpuset file: %v", err)
+	}
+	return cpusList, nil
+}
+
+func GroupCPUThreads(topo CPUTopology) ([]int, error) {
+	cpus, err := topo.GetCPUs()
+	if err != nil {
+		return nil, err
+	}
+
+	cpuSet := NewCPUSet(cpus)
+	res := make([]int, 0, len(cpus))
+
+	appendAndFinish := func(cpu int) bool {
+		res = append(res, cpu)
+		cpuSet.Remove(cpu)
+		return cpuSet.Empty()
```

Comment:
Yes, its okay to delete a map element in a loop. Though here I iterate over the original array `cpus`, not over the map `cpuSet`. The reason beind is that in go maps there is no guarantee about iteration order. AFAIK the iteration order is even randomized hence it will differ from one run to another. Therefore with itaration over a map its not possible to produce stable output. That complicates testability and in general IMHO its not good to provide unpredictable results.